spring-cloud-config-server/src/main/java/org/springframework/cloud/config/server/environment/HttpClientConfigurableHttpConnectionFactory.java

```diff
@@ -105,6 +105,14 @@ private HttpClientBuilder lookupHttpClientBuilder(final URL url) {
 			return HttpClients.custom();
 		}
 		if (builderMap.size() > 1) {
+			List<String> keys = builderMap.keySet().stream().filter(key -> {
```

Comment:
Can you comment here what the goal is?

Reply:
@spencergibb Basically I am determining if there is an exact match URL or not.  So if there is a placeholder in the URL I filter it out.  We should be left with only URLs which have no placeholders.  That is the one we want to use in the case there are multiple matches.